src/mapml.css

```diff
@@ -102,7 +106,6 @@
   border-top: 1px solid transparent;
   border-bottom: 1px solid transparent;
   cursor: default;
-  outline: none;
```

Comment:
Is there a reason not to keep outlines here? This gets overridden:
https://github.com/Maps4HTML/Web-Map-Custom-Element/blob/2edf1b110ab81d7b67c896a8ee7344a3f4ec3dd6/src/mapml.css#L73-L80

Reply:
I had to remove the `overflow-y: auto`, it was causing issues with the submenu, not 100% sure but removing the line seemed to fix it, maybe you might know why?

Reply:
Allowing overflow would enable us to add more options to the menu and not have any of them clipped to the default size of the map element, but I think in most cases developers will increase the default dimensions so probably not an issue.

> not 100% sure but removing the line seemed to fix it, maybe you might know why?

It's not immediately apparent to me why that is, but before digging deeper, there's substantial accessibility issues with the current implementation:

1. The submenu cannot be accessed using a keyboard.
2. The submenu's items come after other primary menu items in the DOM order. So opening a submenu and then tabbing would focus the next primary menu item, as opposed to the first item in that submenu.
3. (Upsteam issue: There's no focus delegation on pressing Shift + F10 to open the context menu, this means that once the context menu has been opened using the keyboard shortcuts, tabbing will focus other map components before the menu items.)

Reply:
It's unfortunate [`<menu>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/menu) was abandoned, or that the plugin [doesn't support submenus](https://github.com/aratcliffe/Leaflet.contextmenu/issues/31).

Would it make the implementation easier if we use the [`<details>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) element (also used in the layer control) where the `open` attribute could be toggled on hovering the menu items that have submenus? It'd require some additional CSS but I think it's doable.

```html
<div class="mapml-contextmenu">
  <!-- primary items goes here -->
  <a class="mapml-contextmenu-item" href="#" role="button">
    Toggle Controls (<kbd>T</kbd>)
  </a>
  ...
  <!-- primary item with submenu -->
  <details>
    <summary>
      <a class="mapml-contextmenu-item" href="#" role="button">
        Copy Coordinates (<kbd>C</kbd>)
      </a>
    </summary>
    <div class="mapml-contextmenu mapml-submenu">
      <!-- submenu items goes here -->
      <a class="mapml-contextmenu-item" href="#" role="button">
        map
      </a>
      ...
    </div>
  </details>

  <!-- primary items goes here -->
</div>
```

Reply:
Would using `<details>` create a dropdown menu? And if so is it possible to make it a popup submenu instead?

Also in relation to accessibility, you are now able to open the submenu using `C`, you do need to tab through all the elements, although this behavior is common it would be ideal to have it focus the context menu, I tried playing around with tab index but I didn't get the desired behavior, any suggestions?

Reply:
It should be possible to make it a popup submenu...

<details><summary>I think</summary>

```html
<style>
.mapml-contextmenu {
  position: absolute;
  width: 150px;
  left: 150px;
}

.mapml-contextmenu details[open] .mapml-submenu {
  position: absolute;
  top: 0;
  width: 120px;
  transform: translate(-120px, 0);
}

/* Remove default disclosure widget (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#Customizing_the_disclosure_widget) */
.mapml-contextmenu summary {
  display: inline-block;
  list-style: none;
}
.mapml-contextmenu ::-webkit-details-marker {
  display: none;
}

/* Presentational, ignore these. */
map {
  contain: content;
  display: inline-block;
  overflow: hidden;
  height: 150px;
  width: 300px;
  border-width: 2px;
  border-style: inset;
}
.mapml-contextmenu,
.mapml-submenu {
  border: 2px dotted black;
}
.mapml-contextmenu-item {
  color: #222;
  cursor: default;
  text-decoration: none;
}
</style>
<map>
  <div class="mapml-contextmenu">
    <a class="mapml-contextmenu-item" href="#" role="button">item 1</a>
    <details class="mapml-contextmenu-item has-mapml-submenu">
      <summary>item 2</summary>
      <div class="mapml-submenu">
        <a class="mapml-contextmenu-item" href="#" role="button">item 2.1</a>
      </div>
    </details>
    <a class="mapml-contextmenu-item" href="#" role="button">item 3</a>
  </div>
</map>
```
</details>

`<details>` is just nice because it has built in accessibility. Otherwise, the [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.2/) has a set of established patterns, such as  [buttons](https://www.w3.org/TR/wai-aria-practices-1.2/#button) (i.e. the menu items `<a role="button">`) and their [recommended keyboard interactions](https://www.w3.org/TR/wai-aria-practices-1.2/#keyboard-interaction-3), and [menu button](https://www.w3.org/TR/wai-aria-practices-1.2/#menubutton) (i.e. the menu items which opens submenus) which might be helpful.